src/components/Cell/Cell.tsx

```diff
@@ -6,17 +6,27 @@ import {flexRender} from '@tanstack/react-table';
 import {getCellClassModes, getCellStyles} from '../../utils';
 import {b} from '../Table/Table.classname';
 
-export interface CellProps<TData> extends React.TdHTMLAttributes<HTMLTableCellElement> {
+export interface CellProps<TData>
+    extends Omit<React.TdHTMLAttributes<HTMLTableCellElement>, 'className'> {
     cell?: CellType<TData, unknown>;
+    className?: string | ((cell: CellType<TData, unknown>) => string);
 }
 
 export const Cell = <TData,>({
     cell,
     children,
-    className,
+    className: classNameProp,
     style,
     ...restProps
 }: CellProps<TData>) => {
+    const className = React.useMemo(() => {
+        if (typeof classNameProp === 'function') {
+            return cell ? classNameProp(cell) : undefined;
```

Comment:
It looks like `cell` should be optional and we should call `classNameProp` even if `cell` is `undefined`.